packages/jest-snapshot/src/SnapshotFile.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const createDirectory = require('jest-util').createDirectory;
+
+const SNAPSHOT_EXTENSION = '.snap';
+const ensureDirectoryExists = filePath => {
+  try {
+    createDirectory(path.join(path.dirname(filePath)));
+  } catch (e) {}
+};
+const fileExists = path => {
```

Comment:
This variant is a bit shorter: https://github.com/facebook/jest/blob/ddbd3beb798f7764dc5917eb8d1ecc6d94c446dd/packages/jest-resolve/src/index.js#L49-L55